text/0000-fix-fees.md

```diff
@@ -0,0 +1,113 @@
+
+# Community-level explanation
+[community-level-explanation]: #community-level-explanation
+
+Grin constrains the size of blocks by a maximum block weight, which is a linear combination of the number of inputs, outputs, and kernels.
+When blocks fill up, miners are incentivized to pick transactions in decreasing order of fees per weight.
+Ideally then, a transaction that is more preferred by miners, is also more likely to be relayed.
+If a transaction A doesn't satisfy the minimum relay fees, while another transaction B does, but miners would rather include A,
+then A's publisher will have to transmit it directly to miners, which is undesirable.
+We therefore want the minimum relay fee be proportional to the weight of the transaction, minimizing arbitrariness.
+We can calibrate the fee system by stipulating that creating one output should cost at least one Grin-cent (formerly 0.4 Grin-cent).
+Linearity is another desirable property, which for instance allows the two parties in a payjoin to each pay their own input and output fees,
+while splitting the kernel fee.
+The least significant bits will specify a minimum fee overpayment factor, preventing aggregation with lesser overpaying transactions.
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+The minimum relay fee of a transaction shall be proportional to Transaction::weight\_as\_block,
+which uses weights of BLOCK\_INPUT\_WEIGHT = 1, BLOCK\_OUTPUT\_WEIGHT = 21, and BLOCK\_KERNEL\_WEIGHT = 3,
+which correspond to the nearest multiple of 32 bytes that it takes to serialize.
+Formerly, we used Transaction::weight,
+which uses arbitrary weights of -1, 4, and 1 respectively, but also non-linearly rounds negative results up to 0
+(as happens when the number of inputs exceeds the number of kernels plus 4 times the number of outputs).
+The Transaction::weight\_as\_block shall be multiplied by a base fee.
+This will not be hardcoded, but configurable in grin-server.toml.
+The already present accept\_fee\_base parameter appears suitable for this, as
+there is no reason to use different fees for relay and mempool acceptance. Its
+value shall default to GRIN\_BASE / 100 / 20 = 500000, which make each output
+incur just over 1 Grin-cent in fees.
+We fix some parameter FEE\_FACTOR\_BITS, possibly configured through
+grin-server.toml.  If the least significant FEE\_FACTOR\_BITS of a fee have
+value f, then the total fees of the transaction containining this kernel must
+exceed the required minimum by a factor f+1, inclusive.
+We recommend a default value of FEE\_FACTOR\_BITS = 10.
+The new tx relay rules and new fee computation in wallets shall take effect at the HF4 block height of 1048320 (but see below about alternatives for 3rd party wallets).
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+I think there is only a perceived drawback, which is what led to the former fee rules.
```

Comment:
Rather than 

> I think there is only a perceived drawback

perhaps 

> At the time of writing, only a single perceived drawback could be identified

or similar?